addons/account/views/account_move_views.xml

```diff
@@ -925,7 +926,7 @@
                                        readonly="state != 'draft'"/>
 
                                 <field name="date" string="Accounting Date"
-                                       invisible="move_type in ('out_invoice', 'out_refund', 'out_receipt') and not quick_edit_mode"
+                                       invisible="move_type in ('out_invoice', 'out_refund', 'out_receipt') and not quick_edit_mode and not show_accounting_date_when_posted"
```

Comment:
I think that to avoid having a computed field, let's put everything in the invisible directly, also since the condition on the move type is already there let's not add it twice :smile: